src/main/kotlin/com/getkeepsafe/dexcount/SourceFile.kt

```diff
@@ -111,11 +111,27 @@ internal class DexFile(
         @JvmStatic
         private fun extractDexFromAar(file: File, dxTimeoutSecs: Int): List<DexFile> {
             // unzip classes.jar from the AAR
+            var minSdk = 13 // the default minSdkVersion of dx
             val tempClasses = file.unzip { entries ->
-                val jarFile = entries.find { it.name.matches(Regex("classes\\.jar")) }
-                makeTempFile("classes.jar").also {
-                    jarFile!!.writeTo(it)
+                var tempFile: File? = null
+
+                for (entry in entries) {
+                    if (entry.name == "AndroidManifest.xml") {
```

Comment:
Correct me if I'm wrong, but this would only work if the minSdk is defined in the manifest rather than in the build.gradle(.kts) where it is often defined, right?

Reply:
Build files (build.gradle and build.gradle.kts) are used to assemble an `.aar` archive, but are not included in it.  The values you define in those build files are _placed into_ the manifest file, where we can reliably read them.

CI is failing here because evidently, the version of `dx` used in CI doesn't have the `--min-sdk-version` option, so now I need to figure out when that flag was added, and conditionally use that option only if it exists.

Reply:
If you doubt this, check our integration tests.  In `integration/lib/build.gradle` you'll certainly find the `minSdkVersion` defined there, and not in `./src/main/AndroidManifest.xml`.

Reply:
Ah, excellent point - I'm not terribly used to working with pre-built archives so I overlooked that. I'll take a look at the new build you published and see if it works. Thanks for being so quick to put in this fix!